implement skip duplicated items

levintech opened the pull request:
now this script generate COUNT items then remove duplicated ones. so the final generated items are less than COUNT.
I fixed this issue to generate correct number of COUNT items that are unique.

levintech committed:
implement skip duplicated items

levintech committed:
added git ignore file

ykeremekinci commented:
Traceback (most recent call last):
  File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 259, in <module>
    main()
  File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 250, in main
    rt = generate_images(edition_name, num_avatars)
  File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 186, in generate_images
    if is_dupliacted(rarity_dict, trait_sets, image_index):
  File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 151, in is_dupliacted
    if rarity_dict[CONFIG[idx]['name']][index] != trait[: -1 * len('.png')]:
TypeError: 'NoneType' object is not subscriptable


I got this error. Can you help me

ykeremekinci reviewed (changes requested):
Traceback (most recent call last):
File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 259, in
main()
File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 250, in main
rt = generate_images(edition_name, num_avatars)
File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 186, in generate_images
if is_dupliacted(rarity_dict, trait_sets, image_index):
File "D:\Google Indirilenler\generative-art-nft-master\generative-art-nft-master\nft.py", line 151, in is_dupliacted
if rarity_dict[CONFIG[idx]['name']][index] != trait[: -1 * len('.png')]:
TypeError: 'NoneType' object is not subscriptable

I got this error. Can you help me

levintech commented:
Hi, @ykeremekinci  Good Day!
Please share me more details....
You can fork this repository and create new branch. Once you push your own branch to this repo, let me know it's name.
I will test your script on my side then we will find way to solve above problem.
Thanks

ykeremekinci commented:
https://github.com/ykeremekinci/generative-art-nft/blob/Generative-Art-Nft-problem-ykeremekinci/generative-art-nft-master%20problem.zip

This is my file.  You can download here. Inside the generative-art-nft folder, there is another generative-art-nft folder. Open this in Visual Studio Code.

levintech committed:
added checking None traits feature

levintech committed:
added requirements

levintech commented:
Hi, @ykeremekinci Thanks for sharing your script.
I found what the issue and fixed them. I missed checking None traits mechanism on the script. I had already fixed that issue and test them on my side. You can see the updated code now. If you have any other problem, please kindly share me.
Thanks.

ykeremekinci commented:
Starting task...
| |        #                                                                                                                                                                                    | 15 Elapsed Time: 0:00:00Generated 16 images, 12 are distinct
Removing 4 images...
Saving metadata...
Task complete!
| |         #                                                                     

This time the code works but it deletes the same images as before and there are fewer images when there should have been 16 images.

ykeremekinci commented:
Thank you for your help.

I got a new problem :(

Starting task...
\ |      #                                                                                                                                                                                             | 14 Elapsed Time: 0:00:00Generated 15 images, 11 are distinct
Removing 4 images...
Saving metadata...
Task complete!
| |       #                                                                                                      

This time the code works but it deletes the same images as before and there are fewer images when there should have been 15 images. And yes, I have installed the versions of the plugins in the suggested post you added.

Here is my file : https://github.com/ykeremekinci/generative-art-nft/blob/Generative-Art-Nft-problem-ykeremekinci/generative-art-nft-problem2.rar

akhater committed:
Update nft.py

The append in line 203 sets it to 'none' so I believe that what we shld be comparing it with

that's why the code wasn't finding any duplicates

levintech committed:
Merge pull request #2 from akhater/patch-2

Update nft.py

ykeremekinci commented:
Yes, it really worked this time. Thank you so much 👍

jayresolf132 commented on nft.py:
```diff
@@ -141,14 +141,32 @@ def generate_trait_set_from_config():
+    if count == 0:
+        return False
+
+    for index in range(count):
```
ye
